hybrasyl/Monolith.cs

```diff
@@ -138,6 +143,155 @@ private static void SpawnMonster(Monster monster, Map map)
+                    {
+                        foreach (var importTable in importSet.Table)
+                        {
+                            for (int i = 0; i < importTable.Rolls; i++)
+                            {
+                                if (importTable.Chance >= _rng.NextDouble())
+                                {
+                                    spawnLootTable.Add(ConvertLootTable(importTable));
+                                    continue;
+                                }
+                            }
+                        }
+                    }
+                }
+
+            }
+            return spawnLootTable;
+        }
+
+        /// <summary>
+        /// Used to Convert Hybrasyl.LootTable to Creatures.LootTable
+        /// </summary>
+        /// <param name="lootTableToConvert">Hybrasyl.LootTable to Convert</param>
+        /// <returns>Creatures.LootTable</returns>
+        private Creatures.LootTable ConvertLootTable(Hybrasyl.LootTable lootTableToConvert)
+        {
+            var newLootTable = new Creatures.LootTable();
+            newLootTable.Items = new Creatures.LootTableItemList();
+            newLootTable.Items.Items = new List<Creatures.LootItem>();
+
+            newLootTable.Chance = lootTableToConvert.Chance;
+            newLootTable.Gold.Min = lootTableToConvert.Gold.Min;
+            newLootTable.Gold.Max = lootTableToConvert.Gold.Max;
+            newLootTable.Items.Chance = lootTableToConvert.Items.Chance;
+
+            foreach(var items in lootTableToConvert.Items.Items)
+            {
+                var newLootItem = new Creatures.LootItem();
+                newLootItem.Always = items.Always;
+                newLootItem.Max = items.Max;
+                newLootItem.Min = items.Min;
+                newLootItem.Unique = items.Unique;
+                newLootItem.Value = items.Value;
+                newLootItem.Variants = items.Variants;
+
+                newLootTable.Items.Items.Add(newLootItem);
+            }
+            
+            newLootTable.Items.Rolls = lootTableToConvert.Items.Rolls;
+            newLootTable.Rolls = lootTableToConvert.Rolls;
+
+            return newLootTable;
+        }
+
+        /// <summary>
+        /// Calculate the total amount of lootable gold based on the base amount plus any additional from the loot table(s).
+        /// </summary>
+        /// <returns>Gold uint</returns>
+        public uint LootableGold()
+        {
+            uint lootableGold = 0;
+            var spawnLootGold = _spawn.Loot.Gold;
+
+            if (spawnLootGold != null)
+            {
+                lootableGold += ((uint)_rng.Next((int)(spawnLootGold?.Min ?? 0), (int)(spawnLootGold?.Max ?? 0)));
+            }
+            if (_spawn.Loot.Table != null || _spawn.Loot.Set != null)
+            {
+                _spawnLootTable.ForEach(lootTable => lootableGold += ((uint)_rng.Next((int)(lootTable.Gold?.Min ?? 0), (int)(lootTable.Gold?.Max ?? 0))));
```

Comment:
This seems very strange to me. Why are you checking if a different data structure is null (_spawn.Loot.Table) and then using ForEach on a pregenerated (_spawnLootTable) one? Just use `foreach` on the output from CreateSpawnLootTable. In the event that there is no table or sets, the list should be returned as empty.

Reply:
Will remove. This was part of a different design, so as noted isn't needed now.